### PR TITLE
Login & Register & logout & change password RestAPI

### PR DESCRIPTION
**Pull request by wenliangchen**
import the Django-allauth library and create the localApp accounts to implement the login register change password function backend with Django-allauth
As I have mentioned before in #27 I have implemented the Login & Register & logout & change password functions.(Only the backend)

- [x] import the Django-allauth library
- [x] create the localApp accounts
- [x] login 
- [x] logout 
- [x] register
- [x] change password

### CODE REVIEW

**Comment by Windsooon** [comment]
IIUC, the branch `Login_Rest` should be created after commit fc297f4 so Files changed would not show the different for this commit which is belongs to PR 33 @wenliangchen

**Comment by wenliangchen** [comment]
> IIUC, the branch `Login_Rest` should be created after commit fc297f4 so Files changed would not show the different for this commit which is belongs to PR 33 @wenliangchen

What I did is what you are looking for.

**Comment by Windsooon** [comment]
@wenliangchen, I guess you follow the practice from https://ordinarycoders.com/blog/article/django-user-register-login-logout

IMO, another way to implement this would be 

1. Use https://bulma.io/documentation/form/general/ to generate the form for the front end, but right now we can ignore the style. 
2. Follow the docs like https://docs.djangoproject.com/en/4.1/topics/auth/default/#how-to-log-a-user-in, to signup, login and logout user without use django form.

File: EX/accounts/views.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from allauth.account.decorators import login_required
+from django.http import HttpResponse
+
+
+@login_required
+def home(request):
```
[the remark this answers is not in the document]
Comment (wenliangchen) [reply EX/accounts/views.py:6]: Yes, other functions have not been realized. The only thing here is about using allAuth to realize the login and registration function.

File: EX/accounts/apps.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+class AccountsConfig(AppConfig):
+    name = 'EX.accounts'
+
```
Comment (Windsooon) [inline EX/accounts/apps.py:5]: Delete extra empty line

File: EX/accounts/tests.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (Windsooon) [inline EX/accounts/tests.py:3]: Delete this line

**Comment by Windsooon** [comment]
Please update your commit based on the review. I think we need to add UserManager like https://docs.djangoproject.com/en/4.1/topics/auth/customizing/#a-full-example to make sure user should sign up vir email.

File: requirements.txt [hunk 4]
```diff
@@ -5,3 +5,4 @@ djangorestframework==3.14.0
 Markdown==3.4.1
 pytz==2022.7
 sqlparse==0.4.3
+django-allauth==0.52.0
```
Comment (Windsooon) [inline requirements.txt:8]: This is not related to this PR. We can all Django all auto and set it up in other PR

File: EX/urls.py [hunk 5]
```diff
@@ -14,8 +14,10 @@
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
 from django.contrib import admin
-from django.urls import path
-
+from django.urls import path, include
+from EX.accounts.views import home as account_home
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('accounts/', include('allauth.urls')), # import account change urls
```
Comment (Windsooon) [inline EX/urls.py:21]: This is not related to this PR. We can all Django all auto and set it up in other PR

File: EX/settings.py [hunk 6]
```diff
@@ -121,3 +132,9 @@
 # https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field
 
 DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+
+LOGIN_URL = '/accounts/login'
+LOGIN_REDIRECT_URL = '/accounts/home'
+AUTH_USER_MODEL = 'accounts.UserModel'
+EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
```
Comment (Windsooon) [inline EX/settings.py:139]: Is this for sending a email?
